web/tsconfig.base.json

```diff
@@ -11,14 +11,6 @@
     "module": "node16",
     "target": "es5",
 
-    // TODO: These override ../tsconfig.base.json settings, and so should be removed if possible,
-    //       but existing code in web/ breaks some of these settinsg
-    "noImplicitThis": false,
-    "noImplicitReturns": false,
-    "noImplicitAny": false,
-    "strictFunctionTypes": false,
-    "noUnusedLocals": false,
```

Comment:
The big motivator:  cleaning this up.  I was able to push it back significantly outside of web/ (and common/web/gesture-recognizer).

Reply:
The latter:  any TS compilation errors encountered by `mocha` when loading tests are not passed through, leaving the cause somewhat opaque.

I resolve these over the following PRs, one flag at a time, along with the corresponding settings in web/.